_posts/2024/2024-11-12-Cobbler-v3.3.6-Beginners-Guide.md

```diff
@@ -0,0 +1,560 @@
+> 
+> - [serverfault](https://serverfault.com/questions/191390/iptables-and-dhcp-questions/) & [stackexchange](https://unix.stackexchange.com/questions/447440/ufw-iptables-not-blocking-dhcp-udp-port-67) sources.
+
+Reconfigure **enp0s8** to operate in the **new firewalld zone**
+
+```shell
+nmcli con mod enp0s8 connection.zone cobbler
+```
+
+> As defined above, this zone will NOT permit inbound ICMP traffic like ping
+
+For simplicity, **enp0s3** can be left in the **default firewalld zone** (verify the current active zones with `firewall-cmd --get-active-zones`)
+
+#### Service confinement
+
+On the topic of network security, we should also restrict our Cobbler services to only listen on the PXE server interface.
+
+- configure HTTP to only listen on the PXE server interface:
+
+```shell
+sed -i 's,^Listen.*,Listen 10.0.0.10:80,' /etc/httpd/conf/httpd.conf
+```
+
+ - Create a local systemd service for tftpd, modifiedying the original config to only listen on a specific IP address:
+ 
+```shell
+systemctl stop tftp.service
+systemctl stop tftp.socket
+
+cp /lib/systemd/system/tftp.service /etc/systemd/system/.
+sed 's,ListenDatagram=69,ListenDatagram=10.0.0.10:69,' /lib/systemd/system/tftp.socket | tee /etc/systemd/system/tftp.socket
+```
+
+- now, reload the systemd configuration and restart services
+
+```shell
+systemctl daemon-reload
+systemctl restart tftp httpd
+```
+
+> Feel free to comment with additional security hardening measures you would employ.
+
+## Tips & Troubleshooting
+
+1. Cobbler's [official docs](https://cobbler.readthedocs.io/en/v3.3.6/installation-guide.html#id2) advise that installing and running Cobbler in a virtual environment is not possible, but I have had no issues related to virtualization.
+
+1. Some dependencies are missing from the official [Cobbler docs](https://cobbler.readthedocs.io/en/v3.3.6/installation-guide.html#packages). Have a look at the [dockerfile](https://github.com/cobbler/cobbler/blob/v3.3.6/docker/rpms/Fedora_34/Fedora34.dockerfile) located in cobbler's source code for a more complete list.
+
+1. It's always a good idea to manually install the desired distro (Fedora 34 or 37 Server in the case of this guide) to the desired hardware (VirtualBox VM in this case) before attempting a network installation, to validate compatability.
+
+    - Any Fedora installation, whether installed manually or automatically, should always contain the file `/root/anaconda-ks.cfg`, which is the kickstart file used to install that particular system.
+
+    - Depending on your deployed system of choice (Fedora 34 or 37 Server here) if you are having issues, you may want to familiarize yourself with the installation system environment.
+
+        1. With most distributions, while the installer is progressing and outputing logs to the monitor, you can press **ALT+F2** or **ALT+F3** or a similar key combo to access a privileged shell to troubleshoot.
```

Comment:
This only works on systems that use ttys. When using a serial console via IPMI sol for example this is not available.